packages/labs/analyzer/src/lib/analyzer.ts

```diff
@@ -73,6 +75,18 @@ export class Analyzer implements AnalyzerInterface {
       ),
     });
   }
+
+  addDiagnostic(options: DiagnosticOptions) {
+    this.diagnostics.push(createDiagnostic(options));
+  }
+
+  *getDiagnostics(): IterableIterator<ts.Diagnostic> {
+    yield* this.diagnostics;
+  }
```

Comment:
This feels like something that someone using the analyzer's API would want but it's only used in tests as of now. Should I update this to avoid putting on the public API for now?